src/NuGet.Core/NuGet.Frameworks/CompatCacheKey.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+
+namespace NuGet.Frameworks
+{
+    /// <summary>
+    /// Internal cache key used to store framework compatibility.
+    /// </summary>
+    internal class CompatCacheKey : IEquatable<CompatCacheKey>
```

Comment:
nit: can we call it `CompatibilityCacheKey`? All the other things have the full word.